server/app/files.go

```diff
@@ -119,20 +127,66 @@ func (a *App) GetFilePath(teamID, rootID, fileName string) (*mm_model.FileInfo,
 	if fileInfo != nil && fileInfo.Path != "" && fileInfo.Path != emptyString {
 		filePath = fileInfo.Path
 	} else {
+		// Validate path components to ensure proper file path handling
+		if err := validatePathComponent(teamID); err != nil {
+			return nil, "", fmt.Errorf("invalid teamID in GetFilePath: %w", err)
+		}
+		if err := validatePathComponent(rootID); err != nil {
+			return nil, "", fmt.Errorf("invalid rootID in GetFilePath: %w", err)
+		}
+		if err := validatePathComponent(fileName); err != nil {
+			return nil, "", fmt.Errorf("invalid fileName in GetFilePath: %w", err)
+		}
 		filePath = filepath.Join(teamID, rootID, fileName)
 	}
 
 	return fileInfo, filePath, nil
 }
 
-func getDestinationFilePath(isTemplate bool, teamID, boardID, filename string) string {
+func getDestinationFilePath(isTemplate bool, teamID, boardID, filename string) (string, error) {
+	// Validate inputs to ensure proper file path handling
+	if err := validatePathComponent(teamID); err != nil {
+		return "", fmt.Errorf("invalid teamID: %w", err)
+	}
+	if err := validatePathComponent(boardID); err != nil {
+		return "", fmt.Errorf("invalid boardID: %w", err)
+	}
+	if err := validatePathComponent(filename); err != nil {
+		return "", fmt.Errorf("invalid filename: %w", err)
+	}
+
 	// if saving a file for a template, save using the "old method" that is /teamID/boardID/fileName
 	// this will prevent template files from being deleted by DataRetention,
 	// which deletes all files inside the "date" subdirectory
 	if isTemplate {
-		return filepath.Join(teamID, boardID, filename)
+		// Use a safe base path for templates with proper validation
+		basePath := filepath.Join(utils.GetBaseFilePath(), "templates")
+		return filepath.Join(basePath, teamID, boardID, filename), nil
 	}
-	return filepath.Join(utils.GetBaseFilePath(), filename)
+	return filepath.Join(utils.GetBaseFilePath(), filename), nil
+}
+
+// validatePathComponent ensures a path component contains only valid characters.
+func validatePathComponent(component string) error {
+	if component == "" {
+		return errEmptyPathComponent
+	}
+
+	if strings.Contains(component, "..") {
+		return errInvalidPathComponent
+	}
+
+	if strings.HasPrefix(component, "/") || strings.HasPrefix(component, "\\") {
+		return errInvalidPathFormat
+	}
```

Comment:
Is this redundant, given the regex below?

Reply:
Removed it. Nice catch 🙌